### PR TITLE
fixes #8591 - explicitly set where to subnets.id

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

File: app/helpers/hosts_and_hostgroups_helper.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def accessible_subnets
   def domain_subnets(domain = @domain)
     return [] if domain.blank?
     ids = domain.subnets.pluck(:id)
-    accessible_subnets.where(:id => ids)
+    accessible_subnets.where('subnets.id' => ids)
```
Comment (ohadlevy) [inline app/helpers/hosts_and_hostgroups_helper.rb:32]: can you add a test please?

**Comment by domcleal** [comment]
Is the error not on line 31, the pluck(:id)?

**Comment by shlomizadok** [comment]
Could happen on both. Changed both (Thanks @domcleal )

**Comment by dLobatog** [comment]
Merged as 395613b415d03bdf82c7a327abad4c9e46fdc99e, thanks @shlomizadok !

**Pull request closed without merging**